manager/allocator/allocator.go

```diff
@@ -817,28 +635,33 @@ func (a *Allocator) processPendingAllocations(ctx context.Context) {
 		// tasks is a map from service id to a slice of task objects belonging
 		// to that service
 		tasks := map[string][]*api.Task{}
-		storeDone := metrics.StartTimer(storeLockHeld.WithValues("service", "allocate", "read"))
 		a.store.View(func(tx store.ReadTx) {
 			for taskid := range pendingTasks {
 				t := store.GetTask(tx, taskid)
-				// if the task is not nil, is in state NEW, and its desired
-				// state is past NEW but not terminal, then the task still
-				// needs allocation.
+				// if the task is nil, then that means it has been deleted
 				if t != nil {
-					if t.Status.State == api.TaskStateNew && api.TaskStateNew < t.DesiredState && t.DesiredState <= api.TaskStateRunning {
-						// if the task has an empty service, then we don't need to
-						// allocate the service. however, we'll keep those
-						// serviceless tasks (network attachment tasks) under the
-						// emptystring key in the map and specialcase that key
-						// later.
-						service := store.GetService(tx, t.ServiceID)
-						if service != nil {
-							services = append(services, service)
+					// only new tasks need to be allocated. However, if a task
```

Comment:
Not gonna lie, this 450 line-function is a bit hard to follow... Might be worthwhile to break it up a little bit?

Reply:
(comment on the whole `processPendingAllocations` func, not just this line)